exercises/src/test/scala/fpinscala/localeffects/LocalEffectsSpec.scala

```diff
@@ -3,5 +3,65 @@ package fpinscala.localeffects
 import org.specs2.mutable._
 
 class LocalEffectsSpec extends Specification {
+  "STArray" >> {
+    "fill" >> {
+
+      val runnable = new RunnableST[(Int, Int)] {
+        override def apply[S]: ST[S, (Int, Int)] = {
+          for {
+            a <- STArray(10, 1)
+            _ <- a.fill(Map(0 -> 5, 5 -> 6))
+            first <- a.read(0)
+            second <- a.read(1)
+          } yield {
+            (first, second)
+          }
+        }
+      }
+
+      ST.runST(runnable) === (5, 1)
+    }
+  }
+
+  "Immutable" >> {
+    "quicksort" >> {
+      def test(xs: List[Int]) = {
+        Immutable.quicksort(xs) === xs.sorted
```

Comment:
It can be made faster by doing a proper check on is sorted rather than compare on a sorted version, but what I did was simply easier and for the small lists being managed shouldn't make any difference.

But your check isn't right though, e.g. `isSorted(List(4, 5, 1, 2))` will be true. Here would be a version that works:

`def isSorted(xs: List[Int]) = (xs zip xs.drop(1)) forall { case (a, b) => a <= b }`